File: class_tool/src/main/java/com/tempsfteam/class_tool/vo/CourseVO.java
package com.tempsfteam.class_tool.vo;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

/**
 * @author hypocodeemia
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
// TODO:等前端接口切换完成，把professionIdsStr和professionIds删除
public class CourseVO {
    private Integer courseId;

    private String name;

    private String icon;

    private String professionIdsStr;

    private List<Integer> professionIds;

    private String description;

    private ArrayList<SimpleExperiment> simpleExperiments;

    private Integer click;

}

File: class_tool/src/main/resources/static/src/models/vo/courseVO.js
export default {
    name: 'courseVO',
    title: 'CourseVO',
    fields: [
        { name: 'courseId', label: '课程ID', type: 'number' },
        { name: 'name', label: '课程名', type: 'string' },
        { name: 'icon', label: '图标', type: 'string' },
        { name: 'professionIdsStr', label: '专业IDs 字符串', type: 'string' },
        { name: 'professionIds', label: '专业IDs', type: 'array' },
        { name: 'description', label: '描述', type: 'string' },
        { name: 'simpleExperiments', label: '实验列表', type: 'array' },
        { name: 'click', label: '点击量', type: 'number' }
    ]
};